sb3_contrib/cem/cem.py

```diff
@@ -0,0 +1,366 @@
+import copy
+import time
+import warnings
+from functools import partial
+from typing import Any, Dict, Optional, Type, Union
+
+import gym
+import numpy as np
+import torch as th
+import torch.nn.utils
+from stable_baselines3.common.base_class import BaseAlgorithm
+from stable_baselines3.common.callbacks import BaseCallback
+from stable_baselines3.common.evaluation import evaluate_policy
+from stable_baselines3.common.policies import BasePolicy
+from stable_baselines3.common.type_aliases import GymEnv, MaybeCallback, Schedule
+from stable_baselines3.common.utils import safe_mean
+from torch.distributions.multivariate_normal import MultivariateNormal
+
+from sb3_contrib.ars.policies import ARSPolicy
+from sb3_contrib.common.vec_env.async_eval import AsyncEval
+
+
+class CEM(BaseAlgorithm):
+    """
+    Noisy Cross Entropy Method: http://dx.doi.org/10.1162/neco.2006.18.12.2936
+    "Learning Tetris Using the Noisy Cross-Entropy Method"
+
+    John Schulman's implementation: https://github.com/joschu/modular_rl/blob/master/modular_rl/cem.py
```

Comment:
2cents, but might be worth to add a link and a mention to evolution strategies here, as this algorithm is so close (or the same) as some under ES names :). Just to help people connect the dots.

https://blog.otoro.net/2017/10/29/visual-evolution-strategies/